Description: Debianize PacketFence
Author: Fabrice Durand <fdurand@inverse.ca>

--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -573,7 +573,7 @@ snort_binary=/usr/sbin/snort
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.dhcpd_binary
 #
@@ -593,12 +593,12 @@ snmptrapd_binary=/usr/sbin/snmptrapd
 # services.radiusd_binary
 # 
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 # 
 # services.arp_binary
 # 
 # Location of the arp binary. Only necessary to change if you are not running the RPMed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 
 [vlan]
 #
--- a/conf/httpd.conf.apache22
+++ b/conf/httpd.conf.apache22
@@ -8,6 +8,7 @@ ServerSignature Off
 UseCanonicalName Off
 Timeout 50
 KeepAliveTimeout 10
+ServerRoot /usr/lib/apache2
 
 # PERFORMANCE RELATED
 # These values are automatically adjusted by PacketFence based on total system memory
