Description: Debianize PacketFence
Author: Fabrice Durand <fdurand@inverse.ca>

--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -463,7 +463,7 @@ suricata_binary=/usr/bin/suricata
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.dhcpd_binary
 #
@@ -478,12 +478,12 @@ snmptrapd_binary=/usr/sbin/snmptrapd
 # services.radiusd_binary
 # 
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 # 
 # services.arp_binary
 # 
 # Location of the arp binary. Only necessary to change if you are not running the RPMed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 # 
 # services.memcached_binary
 # 
