diff --git a/conf/pf.conf.defaults b/conf/pf.conf.defaults
index c2b16ff..518f3a4 100644
--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -413,7 +413,7 @@ radiusd=enabled
 # services.radiusd_binary
 # 
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 #
 # services.iptables
 #
@@ -478,7 +478,7 @@ httpd_graphite=enabled
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.pfsetvlan
 #
@@ -584,7 +584,7 @@ statsd=enabled
 # services.statsd_binary
 # 
 # Location of the statsd (node) binary. Only necessary to change if you are not running the pre-packaged version.
-statsd_binary=/usr/bin/node
+statsd_binary=/usr/bin/nodejs
 #
 # services.radsniff
 #
@@ -604,7 +604,7 @@ p0f=enabled
 # services.p0f_binary
 # 
 # Location of the p0f binary. Only necessary to change if you are not running the pre-packaged version.
-p0f_binary=/usr/local/bin/p0f
+p0f_binary=/usr/sbin/p0f
 #
 # services.openssl_binary
 #
@@ -614,7 +614,7 @@ openssl_binary=/usr/bin/openssl
 # services.arp_binary
 # 
 # location of the arp binary. only necessary to change if you are not running the rpmed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 
 #
 # services.routes
diff --git a/conf/systemd/packetfence-dhcpd.service b/conf/systemd/packetfence-dhcpd.service
index 8f588bc..547c22f 100644
--- a/conf/systemd/packetfence-dhcpd.service
+++ b/conf/systemd/packetfence-dhcpd.service
@@ -12,7 +12,7 @@ StartLimitInterval=60
 Type=notify
 PIDFile=/usr/local/pf/var/run/dhcpd.pid
 ExecStartPre=/usr/local/pf/bin/pfcmd service dhcpd generateconfig
-ExecStartPre=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::dhcpd' -e 'pf::services::manager::dhcpd->new()->preStartSetup()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::dhcpd' -e 'pf::services::manager::dhcpd->new()->preStartSetup()'
 ExecStart=/usr/sbin/dhcpd -pf /usr/local/pf/var/run/dhcpd.pid -f -lf /usr/local/pf/var/dhcpd/dhcpd.leases -cf /usr/local/pf/var/conf/dhcpd.conf 
 Restart=on-failure
 Slice=packetfence.slice
diff --git a/conf/systemd/packetfence-iptables.service b/conf/systemd/packetfence-iptables.service
index f361e07..fac5cc1 100644
--- a/conf/systemd/packetfence-iptables.service
+++ b/conf/systemd/packetfence-iptables.service
@@ -9,8 +9,8 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_start()'
-ExecStop=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
+ExecStart=/usr/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_start()'
+ExecStop=/usr//bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
 Slice=packetfence-base.slice
 
 [Install]
diff --git a/conf/systemd/packetfence-mariadb.service b/conf/systemd/packetfence-mariadb.service
index 5b6b112d..32c9bc3 100644
--- a/conf/systemd/packetfence-mariadb.service
+++ b/conf/systemd/packetfence-mariadb.service
@@ -7,6 +7,7 @@ Wants=packetfence-config.service
 Conflicts=mysql.service mariadb.service
 
 [Service]
+Environment=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 StandardOutput=syslog
 StartLimitBurst=3
 StartLimitInterval=60
diff --git a/conf/systemd/packetfence-routes.service b/conf/systemd/packetfence-routes.service
index 36d7625..3c5e56c 100644
--- a/conf/systemd/packetfence-routes.service
+++ b/conf/systemd/packetfence-routes.service
@@ -9,8 +9,8 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->start()'
-ExecStop=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->stop()'
+ExecStart=/usr/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->start()'
+ExecStop=/usr//bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->stop()'
 Slice=packetfence-base.slice
 
 [Install]
