diff --git a/conf/mariadb/mariadb.conf.tt.example b/conf/mariadb/mariadb.conf.tt.example
index f70bd6f..670ef55 100644
--- a/conf/mariadb/mariadb.conf.tt.example
+++ b/conf/mariadb/mariadb.conf.tt.example
@@ -18,7 +18,7 @@ skip-networking
 bind-address=
 [% END %]
 port    = 3306
-socket    = /var/lib/mysql/mysql.sock
+socket    = /var/run/mysqld/mysqld.sock
 skip-external-locking
 key_buffer_size = [% key_buffer_size %]M
 max_allowed_packet = [% max_allowed_packet %]M
diff --git a/conf/pf.conf.defaults b/conf/pf.conf.defaults
index c2b16ff..518f3a4 100644
--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -413,7 +413,7 @@ radiusd=enabled
 # services.radiusd_binary
 # 
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 #
 # services.iptables
 #
@@ -478,7 +478,7 @@ httpd_graphite=enabled
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.pfsetvlan
 #
@@ -584,7 +584,7 @@ statsd=enabled
 # services.statsd_binary
 # 
 # Location of the statsd (node) binary. Only necessary to change if you are not running the pre-packaged version.
-statsd_binary=/usr/bin/node
+statsd_binary=/usr/bin/nodejs
 #
 # services.radsniff
 #
@@ -604,7 +604,7 @@ p0f=enabled
 # services.p0f_binary
 # 
 # Location of the p0f binary. Only necessary to change if you are not running the pre-packaged version.
-p0f_binary=/usr/local/bin/p0f
+p0f_binary=/usr/sbin/p0f
 #
 # services.openssl_binary
 #
@@ -614,7 +614,7 @@ openssl_binary=/usr/bin/openssl
 # services.arp_binary
 # 
 # location of the arp binary. only necessary to change if you are not running the rpmed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 
 #
 # services.routes
diff --git a/conf/systemd/packetfence-dhcpd.service b/conf/systemd/packetfence-dhcpd.service
index b9fdb51..030c89e 100644
--- a/conf/systemd/packetfence-dhcpd.service
+++ b/conf/systemd/packetfence-dhcpd.service
@@ -11,7 +11,7 @@ StartLimitInterval=60
 Type=notify
 PIDFile=/usr/local/pf/var/run/dhcpd.pid
 ExecStartPre=/usr/local/pf/bin/pfcmd service dhcpd generateconfig
-ExecStartPre=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::dhcpd' -e 'pf::services::manager::dhcpd->new()->preStartSetup()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::dhcpd' -e 'pf::services::manager::dhcpd->new()->preStartSetup()'
 ExecStart=/usr/sbin/dhcpd -pf /usr/local/pf/var/run/dhcpd.pid -f -lf /usr/local/pf/var/dhcpd/dhcpd.leases -cf /usr/local/pf/var/conf/dhcpd.conf 
 Restart=on-failure
 Slice=packetfence.slice
diff --git a/conf/systemd/packetfence-iptables.service b/conf/systemd/packetfence-iptables.service
index 8880030..5539442 100644
--- a/conf/systemd/packetfence-iptables.service
+++ b/conf/systemd/packetfence-iptables.service
@@ -7,8 +7,8 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/bin/perl -I/usr/local/pf/lib -Mpf::db -Mpf::services::manager::iptables -e 'my $db ; while(!$db) { eval { $db = db_connect() } } ; pf::services::manager::iptables->new()->_start()'
-ExecStop=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
+ExecStart=/usr/bin/perl -I/usr/local/pf/lib -Mpf::db -Mpf::services::manager::iptables -e 'my $db ; while(!$db) { eval { $db = db_connect() } } ; pf::services::manager::iptables->new()->_start()'
+ExecStop=/usr//bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
 Slice=packetfence-base.slice
 
 [Install]
diff --git a/conf/systemd/packetfence-routes.service b/conf/systemd/packetfence-routes.service
index 5bc3736..f0c7fc4 100644
--- a/conf/systemd/packetfence-routes.service
+++ b/conf/systemd/packetfence-routes.service
@@ -8,8 +8,8 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->start()'
-ExecStop=/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->stop()'
+ExecStart=/usr/bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->start()'
+ExecStop=/usr//bin/perl -I/usr/local/pf/lib '-Mpf::services::manager::routes' -e 'pf::services::manager::routes->new()->stop()'
 Slice=packetfence-base.slice
 
 [Install]
