diff --git a/conf/keepalived.conf.example b/conf/keepalived.conf.example
index 54b2113be5..c96aaba72c 100644
--- a/conf/keepalived.conf.example
+++ b/conf/keepalived.conf.example
@@ -9,7 +9,7 @@ global_defs {
 }
 
 vrrp_track_process radius_load_balancer {
-  process /usr/sbin/radiusd -d /usr/local/pf/raddb -n load_balancer -fm
+  process /usr/sbin/freeradius -d /usr/local/pf/raddb  -n load_balancer -fm
   full_command
   quorum 1
   delay 15
diff --git a/conf/mariadb/mariadb.conf.tt.example b/conf/mariadb/mariadb.conf.tt.example
index 591715aef3..7fc8654caf 100644
--- a/conf/mariadb/mariadb.conf.tt.example
+++ b/conf/mariadb/mariadb.conf.tt.example
@@ -19,7 +19,7 @@ skip-networking
 bind-address=
 [% END %]
 port    = 3306
-socket    = /var/lib/mysql/mysql.sock
+socket    = /var/run/mysqld/mysqld.sock
 skip-external-locking
 key_buffer_size = [% key_buffer_size %]M
 max_allowed_packet = [% max_allowed_packet %]M
diff --git a/conf/monitoring/netdata.conf.example b/conf/monitoring/netdata.conf.example
index d3a3cd9b63..fe284fcaa2 100644
--- a/conf/monitoring/netdata.conf.example
+++ b/conf/monitoring/netdata.conf.example
@@ -8,7 +8,7 @@
 [global]
 	run as user = netdata
 	web files owner = root
-	web files group = netdata
+	web files group = root
         config directory = /usr/local/pf/var/conf/monitoring
         # Netdata is not designed to be exposed to potentially hostile networks
         # See https://github.com/firehol/netdata/issues/164
diff --git a/conf/pf.conf.defaults b/conf/pf.conf.defaults
index d1c03b2cf0..6bd56838ad 100644
--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -429,7 +429,7 @@ radiusd=enabled
 # services.radiusd_binary
 # 
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 #
 # services.radiusd_acct
 #
@@ -494,7 +494,7 @@ httpd_proxy=enabled
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.snmptrapd
 #
@@ -592,7 +592,7 @@ openssl_binary=/usr/bin/openssl
 # services.arp_binary
 # 
 # location of the arp binary. only necessary to change if you are not running the rpmed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 #
 # services.netdata
 #
@@ -610,8 +610,8 @@ tc=enabled
 #
 # services.tc_binary
 #
-# location of the yc binary. only necessary to change if you are not running the rpmed version.
-tc_binary=/usr/sbin/tc
+# location of the tc binary. only necessary to change if you are not running the rpmed version.
+tc_binary=/sbin/tc
 # services.pfdns
 #
 # Should pfdns be managed by PacketFence?
diff --git a/conf/systemd/packetfence-haproxy-admin.service b/conf/systemd/packetfence-haproxy-admin.service
index ae252577a2..c9247ab348 100644
--- a/conf/systemd/packetfence-haproxy-admin.service
+++ b/conf/systemd/packetfence-haproxy-admin.service
@@ -8,7 +8,7 @@ After=packetfence-base.target packetfence-config.service packetfence-pfperl-api.
 Type=notify
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_admin' -e 'pf::services::manager::haproxy_admin->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_admin' -e 'pf::services::manager::haproxy_admin->new()->generateConfig()'
 ExecStart=/usr/sbin/haproxy -Ws -f /usr/local/pf/var/conf/haproxy-admin.conf -p /usr/local/pf/var/run/haproxy-admin.pid
 ExecReload=/bin/kill -USR2 $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-haproxy-db.service b/conf/systemd/packetfence-haproxy-db.service
index b9c09053d9..30735000bb 100644
--- a/conf/systemd/packetfence-haproxy-db.service
+++ b/conf/systemd/packetfence-haproxy-db.service
@@ -8,7 +8,7 @@ Wants=packetfence-config.service
 Type=notify
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_db' -e 'pf::services::manager::haproxy_db->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_db' -e 'pf::services::manager::haproxy_db->new()->generateConfig()'
 ExecStart=/usr/sbin/haproxy -Ws -f /usr/local/pf/var/conf/haproxy-db.conf -p /usr/local/pf/var/run/haproxy-db.pid
 ExecReload=/bin/kill -USR2 $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-haproxy-portal.service b/conf/systemd/packetfence-haproxy-portal.service
index 9d0d61e472..db010f5f41 100644
--- a/conf/systemd/packetfence-haproxy-portal.service
+++ b/conf/systemd/packetfence-haproxy-portal.service
@@ -7,7 +7,7 @@ Wants=packetfence-config.service
 Type=notify
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_portal' -e 'pf::services::manager::haproxy_portal->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_portal' -e 'pf::services::manager::haproxy_portal->new()->generateConfig()'
 ExecStart=/usr/sbin/haproxy -Ws -f /usr/local/pf/var/conf/haproxy-portal.conf -p /usr/local/pf/var/run/haproxy-portal.pid
 ExecReload=/bin/kill -USR2 $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-httpd.aaa.service b/conf/systemd/packetfence-httpd.aaa.service
index 68f5817993..b2f6f4ccba 100644
--- a/conf/systemd/packetfence-httpd.aaa.service
+++ b/conf/systemd/packetfence-httpd.aaa.service
@@ -12,7 +12,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=notify
 ExecStartPre=/usr/bin/rm -f /usr/local/pf/var/run/httpd.aaa.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_aaa' -e 'pf::services::manager::httpd_aaa->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_aaa' -e 'pf::services::manager::httpd_aaa->new()->generateConfig()'
 ExecStart=/usr/sbin/httpd -f /usr/local/pf/var/conf/httpd.conf.d/httpd.aaa -DFOREGROUND  -Drhel
 ExecReload=/bin/kill -USR1 ${MAINPID}
 PrivateTmp=true
diff --git a/conf/systemd/packetfence-httpd.collector.service b/conf/systemd/packetfence-httpd.collector.service
index 59bbf12796..6e64de73e3 100644
--- a/conf/systemd/packetfence-httpd.collector.service
+++ b/conf/systemd/packetfence-httpd.collector.service
@@ -11,7 +11,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=notify
 ExecStartPre=/usr/bin/rm -f /usr/local/pf/var/run/httpd.collector.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_collector' -e 'pf::services::manager::httpd_collector->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_collector' -e 'pf::services::manager::httpd_collector->new()->generateConfig()'
 ExecStart=/usr/sbin/httpd -f /usr/local/pf/var/conf/httpd.conf.d/httpd.collector -DFOREGROUND  -Drhel
 ExecReload=/bin/kill -USR1 ${MAINPID}
 ExecStop=/bin/kill -WINCH ${MAINPID}
diff --git a/conf/systemd/packetfence-httpd.portal.service b/conf/systemd/packetfence-httpd.portal.service
index 325486cacc..06fb9a3cc7 100644
--- a/conf/systemd/packetfence-httpd.portal.service
+++ b/conf/systemd/packetfence-httpd.portal.service
@@ -11,7 +11,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=notify
 ExecStartPre=/usr/bin/rm -f /usr/local/pf/var/run/httpd.portal.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_portal' -e 'pf::services::manager::httpd_portal->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_portal' -e 'pf::services::manager::httpd_portal->new()->generateConfig()'
 ExecStart=/usr/sbin/httpd -f /usr/local/pf/var/conf/httpd.conf.d/httpd.portal -DFOREGROUND  -Drhel
 ExecReload=/bin/kill -USR1 ${MAINPID}
 ExecStop=/bin/kill -WINCH ${MAINPID}
diff --git a/conf/systemd/packetfence-httpd.proxy.service b/conf/systemd/packetfence-httpd.proxy.service
index 8399433b77..cdd5151bc2 100644
--- a/conf/systemd/packetfence-httpd.proxy.service
+++ b/conf/systemd/packetfence-httpd.proxy.service
@@ -11,7 +11,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=notify
 ExecStartPre=/usr/bin/rm -f /usr/local/pf/var/run/httpd.proxy.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_proxy' -e 'pf::services::manager::httpd_proxy->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_proxy' -e 'pf::services::manager::httpd_proxy->new()->generateConfig()'
 ExecStart=/usr/sbin/httpd -f /usr/local/pf/var/conf/httpd.conf.d/httpd.proxy -DFOREGROUND  -Drhel
 ExecReload=/bin/kill -USR1 ${MAINPID}
 PrivateTmp=true
diff --git a/conf/systemd/packetfence-httpd.webservices.service b/conf/systemd/packetfence-httpd.webservices.service
index 2d7b8745b9..2c4956b015 100644
--- a/conf/systemd/packetfence-httpd.webservices.service
+++ b/conf/systemd/packetfence-httpd.webservices.service
@@ -11,7 +11,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=notify
 ExecStartPre=/usr/bin/rm -f /usr/local/pf/var/run/httpd.webservices.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_webservices' -e 'pf::services::manager::httpd_webservices->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_webservices' -e 'pf::services::manager::httpd_webservices->new()->generateConfig()'
 ExecStart=/usr/sbin/httpd -f /usr/local/pf/var/conf/httpd.conf.d/httpd.webservices -DFOREGROUND  -Drhel
 ExecReload=/bin/kill -USR1 ${MAINPID}
 ExecStop=/bin/kill -WINCH ${MAINPID}
diff --git a/conf/systemd/packetfence-iptables.service b/conf/systemd/packetfence-iptables.service
index c147a89edd..4cc581ed51 100644
--- a/conf/systemd/packetfence-iptables.service
+++ b/conf/systemd/packetfence-iptables.service
@@ -7,8 +7,8 @@ After=packetfence-base.target packetfence-config.service
 [Service]
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStart=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 -Mpf::db -Mpf::services::manager::iptables -e 'my $db ; while(!$db) { eval { $db = db_connect() } ; sleep 1 } ; pf::services::manager::iptables->new()->startAndCheck()'
-ExecStop=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
+ExecStart=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 -Mpf::db -Mpf::services::manager::iptables -e 'my $db ; while(!$db) { eval { $db = db_connect() } ; sleep 1 } ; pf::services::manager::iptables->new()->startAndCheck()'
+ExecStop=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
 Slice=packetfence.slice
 
 [Install]
diff --git a/conf/systemd/packetfence-keepalived.service b/conf/systemd/packetfence-keepalived.service
index 5277b4103e..31525c5537 100644
--- a/conf/systemd/packetfence-keepalived.service
+++ b/conf/systemd/packetfence-keepalived.service
@@ -9,7 +9,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=forking
 KillMode=process
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::keepalived' -e 'pf::services::manager::keepalived->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::keepalived' -e 'pf::services::manager::keepalived->new()->generateConfig()'
 ExecStart=/usr/sbin/keepalived -f /usr/local/pf/var/conf/keepalived.conf --pid=/usr/local/pf/var/run/keepalived.pid
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-netdata.service b/conf/systemd/packetfence-netdata.service
index eea70993c2..3876e340cb 100644
--- a/conf/systemd/packetfence-netdata.service
+++ b/conf/systemd/packetfence-netdata.service
@@ -13,7 +13,7 @@ RuntimeDirectoryMode=0775
 ExecStart=/usr/sbin/netdata -P /run/netdata/netdata.pid -D -c /usr/local/pf/var/conf/monitoring/netdata.conf 
 ExecStartPre=/bin/mkdir -p /var/cache/netdata
 ExecStartPre=/bin/chown -R netdata:netdata /var/cache/netdata
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::netdata' -e 'pf::services::manager::netdata->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::netdata' -e 'pf::services::manager::netdata->new()->generateConfig()'
 PermissionsStartOnly=true
 
 TimeoutSec=60
diff --git a/conf/systemd/packetfence-pfcertmanager.service b/conf/systemd/packetfence-pfcertmanager.service
index 217f17e622..b38b62f12c 100644
--- a/conf/systemd/packetfence-pfcertmanager.service
+++ b/conf/systemd/packetfence-pfcertmanager.service
@@ -11,7 +11,7 @@ StartLimitInterval=60
 Type=notify
 WatchdogSec=30s
 Environment=LOG_LEVEL=INFO
-ExecStartPre=-/usr/bin/umount -l /usr/local/pf/conf/certmanager
+ExecStartPre=-/bin/umount -l /usr/local/pf/conf/certmanager
 ExecStart=/usr/local/pf/sbin/pfcertmanager
 Restart=on-failure
 Slice=packetfence.slice
diff --git a/conf/systemd/packetfence-pfdns.service b/conf/systemd/packetfence-pfdns.service
index 583806b8e1..52b9ff7973 100644
--- a/conf/systemd/packetfence-pfdns.service
+++ b/conf/systemd/packetfence-pfdns.service
@@ -10,7 +10,7 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=notify
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfdns' -e 'pf::services::manager::pfdns->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfdns' -e 'pf::services::manager::pfdns->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfdns -conf=/usr/local/pf/var/conf/pfdns.conf
 Restart=on-failure
 Slice=packetfence.slice
diff --git a/conf/systemd/packetfence-pfqueue.service b/conf/systemd/packetfence-pfqueue.service
index ee15924f1c..e16521ef48 100644
--- a/conf/systemd/packetfence-pfqueue.service
+++ b/conf/systemd/packetfence-pfqueue.service
@@ -10,7 +10,7 @@ After=packetfence-base.target packetfence-config.service packetfence-iptables.se
 Type=notify
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue' -e 'pf::services::manager::pfqueue->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue' -e 'pf::services::manager::pfqueue->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfqueue
 User=pf
 Restart=on-failure
diff --git a/conf/systemd/packetfence-radiusd-acct.service b/conf/systemd/packetfence-radiusd-acct.service
index e900d88bee..9ed67e9a37 100644
--- a/conf/systemd/packetfence-radiusd-acct.service
+++ b/conf/systemd/packetfence-radiusd-acct.service
@@ -12,7 +12,7 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
 ExecStartPre=/usr/local/pf/bin/pfcmd service radiusd generateconfig
 ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n acct -Cxm -lstdout 
 ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n acct -fm 
diff --git a/conf/systemd/packetfence-radiusd-auth.service b/conf/systemd/packetfence-radiusd-auth.service
index fa93bf1c5c..9b9bb9ad7a 100644
--- a/conf/systemd/packetfence-radiusd-auth.service
+++ b/conf/systemd/packetfence-radiusd-auth.service
@@ -14,7 +14,7 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
 ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n auth -Cxm -lstdout 
 ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n auth -fm 
 Restart=on-failure
diff --git a/conf/systemd/packetfence-radiusd-cli.service b/conf/systemd/packetfence-radiusd-cli.service
index ee1c93b266..1ddc48321a 100644
--- a/conf/systemd/packetfence-radiusd-cli.service
+++ b/conf/systemd/packetfence-radiusd-cli.service
@@ -12,7 +12,7 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
 ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n cli -Cxm -lstdout 
 ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n cli -fm 
 Restart=on-failure
diff --git a/conf/systemd/packetfence-radiusd-eduroam.service b/conf/systemd/packetfence-radiusd-eduroam.service
index 14288f8464..e906585ce6 100644
--- a/conf/systemd/packetfence-radiusd-eduroam.service
+++ b/conf/systemd/packetfence-radiusd-eduroam.service
@@ -11,7 +11,7 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
 ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n eduroam -Cxm -lstdout
 ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n eduroam -fm 
 Restart=on-failure
diff --git a/conf/systemd/packetfence-radiusd-load_balancer.service b/conf/systemd/packetfence-radiusd-load_balancer.service
index 565f08c0a4..844ce6339f 100644
--- a/conf/systemd/packetfence-radiusd-load_balancer.service
+++ b/conf/systemd/packetfence-radiusd-load_balancer.service
@@ -12,7 +12,7 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
 ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n load_balancer -Cxm -lstdout
 ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n load_balancer -fm 
 Restart=on-failure
diff --git a/conf/systemd/packetfence-redis_ntlm_cache.service b/conf/systemd/packetfence-redis_ntlm_cache.service
index 04bcced7b5..8e735efcb0 100644
--- a/conf/systemd/packetfence-redis_ntlm_cache.service
+++ b/conf/systemd/packetfence-redis_ntlm_cache.service
@@ -7,7 +7,7 @@ After=packetfence-base.target packetfence-config.service packetfence-iptables.se
 [Service]
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_ntlm_cache' -e 'pf::services::manager::redis_ntlm_cache->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_ntlm_cache' -e 'pf::services::manager::redis_ntlm_cache->new()->generateConfig()'
 ExecStart=/usr/bin/redis-server /usr/local/pf/var/conf/redis_ntlm_cache.conf --daemonize no
 ExecStop=/bin/kill $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-redis_queue.service b/conf/systemd/packetfence-redis_queue.service
index b648a3a9d7..70d0af1ac3 100644
--- a/conf/systemd/packetfence-redis_queue.service
+++ b/conf/systemd/packetfence-redis_queue.service
@@ -7,7 +7,7 @@ After=packetfence-base.target packetfence-config.service packetfence-iptables.se
 [Service]
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_queue' -e 'pf::services::manager::redis_queue->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_queue' -e 'pf::services::manager::redis_queue->new()->generateConfig()'
 ExecStart=/usr/bin/redis-server /usr/local/pf/var/conf/redis_queue.conf --daemonize no
 ExecStop=/bin/kill $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-snmptrapd.service b/conf/systemd/packetfence-snmptrapd.service
index 707f9cba20..40fa1a18c1 100644
--- a/conf/systemd/packetfence-snmptrapd.service
+++ b/conf/systemd/packetfence-snmptrapd.service
@@ -8,7 +8,7 @@ Wants=packetfence-base.target packetfence-config.service packetfence-iptables.se
 StartLimitBurst=3
 StartLimitInterval=10
 Type=notify
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::snmptrapd' -e 'pf::services::manager::snmptrapd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::snmptrapd' -e 'pf::services::manager::snmptrapd->new()->generateConfig()'
 ExecStart=/usr/sbin/snmptrapd -f -n -c /usr/local/pf/var/conf/snmptrapd.conf -C -A -Lf /usr/local/pf/logs/snmptrapd.log -p /usr/local/pf/var/run/snmptrapd.pid -On 
 ExecReload=/bin/kill -HUP $MAINPID
 Slice=packetfence.slice
diff --git a/conf/systemd/packetfence-tc.service b/conf/systemd/packetfence-tc.service
index c3c86ed225..360cd0de29 100644
--- a/conf/systemd/packetfence-tc.service
+++ b/conf/systemd/packetfence-tc.service
@@ -9,8 +9,8 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 -Mpf::db -Mpf::services::manager::tc -e 'my $db ; while(!$db) { eval { $db = db_connect() } ; sleep 1 } ; pf::services::manager::tc->new()->_start()'
-ExecStop=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::tc' -e 'pf::services::manager::tc->new()->_stop()'
+ExecStart=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 -Mpf::db -Mpf::services::manager::tc -e 'my $db ; while(!$db) { eval { $db = db_connect() } ; sleep 1 } ; pf::services::manager::tc->new()->_start()'
+ExecStop=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::tc' -e 'pf::services::manager::tc->new()->_stop()'
 Slice=packetfence.slice
 
 [Install]
