diff --git a/conf/pf.conf.defaults b/conf/pf.conf.defaults
index cba215d..4013600 100644
--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -388,7 +388,7 @@ radiusd_order=12
 # services.radiusd_binary
 # 
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 #
 # services.iptables
 #
@@ -498,7 +498,7 @@ httpd_graphite_order=99
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.pfsetvlan
 #
@@ -670,7 +670,7 @@ statsd_order=93
 # services.statsd_binary
 # 
 # Location of the statsd (node) binary. Only necessary to change if you are not running the pre-packaged version.
-statsd_binary=/usr/bin/node
+statsd_binary=/usr/bin/nodejs
 #
 # services.radsniff
 #
@@ -700,7 +700,7 @@ p0f_order=81
 # services.p0f_binary
 # 
 # Location of the p0f binary. Only necessary to change if you are not running the pre-packaged version.
-p0f_binary=/usr/local/bin/p0f
+p0f_binary=/usr/sbin/p0f
 #
 # services.openssl_binary
 #
@@ -710,7 +710,7 @@ openssl_binary=/usr/bin/openssl
 # services.arp_binary
 # 
 # location of the arp binary. only necessary to change if you are not running the rpmed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 
 #
 # services.routes
