diff --git a/conf/pf.conf.defaults b/conf/pf.conf.defaults
index 7735164bfe..be3978df0a 100644
--- a/conf/pf.conf.defaults
+++ b/conf/pf.conf.defaults
@@ -489,7 +489,7 @@ radiusd=enabled
 # services.radiusd_binary
 #
 # Location of the named binary. Only necessary to change if you are not running the RPMed version.
-radiusd_binary=/usr/sbin/radiusd
+radiusd_binary=/usr/sbin/freeradius
 #
 # services.radiusd_acct
 #
@@ -554,7 +554,7 @@ httpd_aaa=enabled
 # services.httpd_binary
 #
 # Location of the apache binary. Only necessary to change if you are not running the RPMed version.
-httpd_binary=/usr/sbin/httpd
+httpd_binary=/usr/sbin/apache2
 #
 # services.snmptrapd
 #
@@ -678,7 +678,7 @@ openssl_binary=/usr/bin/openssl
 # services.arp_binary
 #
 # location of the arp binary. only necessary to change if you are not running the rpmed version.
-arp_binary=/sbin/arp
+arp_binary=/usr/sbin/arp
 #
 # services.netdata
 #
diff --git a/conf/systemd/packetfence-api-frontend.service b/conf/systemd/packetfence-api-frontend.service
index 61b0f64dce..fc68803028 100644
--- a/conf/systemd/packetfence-api-frontend.service
+++ b/conf/systemd/packetfence-api-frontend.service
@@ -11,15 +11,15 @@ TimeoutStopSec=60
 NotifyAccess=all
 LimitNOFILE=infinity
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::api_frontend' -e 'pf::services::manager::api_frontend->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::api_frontend' -e 'pf::services::manager::api_frontend->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/api-frontend-docker-wrapper
 ExecStop=/bin/bash -c "docker stop api-frontend ; echo Stopped"
 Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/api-frontend-systemd-notify.pid
 # iptables rules
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_api_frontend_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_api_frontend_rules("REMOVE")'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_api_frontend_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_api_frontend_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-galera-autofix.service b/conf/systemd/packetfence-galera-autofix.service
index 83845dd15b..1e940135ac 100644
--- a/conf/systemd/packetfence-galera-autofix.service
+++ b/conf/systemd/packetfence-galera-autofix.service
@@ -12,8 +12,8 @@ User=root
 # Prevents systemd on giving up on the restart of this service
 StartLimitInterval=0
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_galera_autofix_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_galera_autofix_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_galera_autofix_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_galera_autofix_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence-base.target
diff --git a/conf/systemd/packetfence-haproxy-admin.service b/conf/systemd/packetfence-haproxy-admin.service
index 185d2a0661..689b489833 100644
--- a/conf/systemd/packetfence-haproxy-admin.service
+++ b/conf/systemd/packetfence-haproxy-admin.service
@@ -9,7 +9,7 @@ Type=notify
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_admin' -e 'pf::services::manager::haproxy_admin->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_admin' -e 'pf::services::manager::haproxy_admin->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/haproxy-admin-docker-wrapper
 ExecStop=/bin/bash -c "docker stop haproxy-admin ; echo Stopped"
 Restart=on-failure
@@ -17,8 +17,8 @@ TimeoutStopSec=30
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/haproxy-admin-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_admin_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_admin_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_admin_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_admin_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-haproxy-db.service b/conf/systemd/packetfence-haproxy-db.service
index dac2e493ec..5f831702bc 100644
--- a/conf/systemd/packetfence-haproxy-db.service
+++ b/conf/systemd/packetfence-haproxy-db.service
@@ -8,15 +8,15 @@ Wants=packetfence-config.service
 Type=notify
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_db' -e 'pf::services::manager::haproxy_db->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_db' -e 'pf::services::manager::haproxy_db->new()->generateConfig()'
 ExecStart=/usr/sbin/haproxy -Ws -f /usr/local/pf/var/conf/haproxy-db.conf -p /usr/local/pf/var/run/haproxy-db.pid
-ExecReload=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_db' -e 'pf::services::manager::haproxy_db->new()->generateConfig()'
+ExecReload=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_db' -e 'pf::services::manager::haproxy_db->new()->generateConfig()'
 ExecReload=/bin/kill -USR2 $MAINPID
 Restart=on-failure
 SuccessExitStatus=143
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_db_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_db_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_db_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_db_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence-cluster.target
diff --git a/conf/systemd/packetfence-haproxy-portal.service b/conf/systemd/packetfence-haproxy-portal.service
index e134c94bfe..8300722e02 100644
--- a/conf/systemd/packetfence-haproxy-portal.service
+++ b/conf/systemd/packetfence-haproxy-portal.service
@@ -8,7 +8,7 @@ Type=notify
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_portal' -e 'pf::services::manager::haproxy_portal->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::haproxy_portal' -e 'pf::services::manager::haproxy_portal->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/haproxy-portal-docker-wrapper
 ExecStop=/bin/bash -c "docker stop haproxy-portal ; echo Stopped"
 Restart=on-failure
@@ -16,10 +16,10 @@ TimeoutStopSec=30
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/haproxy-portal-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_portal_rules("add")'
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::ip6tables -Mpf::config -e 'pf::ip6tables::ip6tables_haproxy_portal_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_portal_rules("REMOVE")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::ip6tables -Mpf::config -e 'pf::ip6tables::ip6tables_haproxy_portal_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_portal_rules("add")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::ip6tables -Mpf::config -e 'pf::ip6tables::ip6tables_haproxy_portal_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_haproxy_portal_rules("REMOVE")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::ip6tables -Mpf::config -e 'pf::ip6tables::ip6tables_haproxy_portal_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-httpd.aaa.service b/conf/systemd/packetfence-httpd.aaa.service
index 1cecf351cf..ce14457019 100644
--- a/conf/systemd/packetfence-httpd.aaa.service
+++ b/conf/systemd/packetfence-httpd.aaa.service
@@ -12,7 +12,7 @@ Type=notify
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_aaa' -e 'pf::services::manager::httpd_aaa->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_aaa' -e 'pf::services::manager::httpd_aaa->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/httpd.aaa-docker-wrapper
 ExecStop=/bin/bash -c "docker stop httpd.aaa ; echo Stopped"
 TimeoutStopSec=30
@@ -20,8 +20,8 @@ Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/httpd.aaa-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_aaa_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_aaa_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_aaa_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_aaa_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-httpd.admin_dispatcher.service b/conf/systemd/packetfence-httpd.admin_dispatcher.service
index 44cfbbdc44..5441803638 100644
--- a/conf/systemd/packetfence-httpd.admin_dispatcher.service
+++ b/conf/systemd/packetfence-httpd.admin_dispatcher.service
@@ -12,7 +12,7 @@ TimeoutStopSec=60
 NotifyAccess=all
 LimitNOFILE=8192
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_admin_dispatcher' -e 'pf::services::manager::httpd_admin_dispatcher->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_admin_dispatcher' -e 'pf::services::manager::httpd_admin_dispatcher->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/httpd.admin_dispatcher-docker-wrapper
 ExecStop=/bin/bash -c "docker stop httpd.admin_dispatcher ; echo Stopped"
 Restart=on-failure
diff --git a/conf/systemd/packetfence-httpd.dispatcher.service b/conf/systemd/packetfence-httpd.dispatcher.service
index 41046c0882..9ffa2615d8 100644
--- a/conf/systemd/packetfence-httpd.dispatcher.service
+++ b/conf/systemd/packetfence-httpd.dispatcher.service
@@ -11,15 +11,15 @@ TimeoutStopSec=60
 NotifyAccess=all
 LimitNOFILE=8192
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_dispatcher' -e 'pf::services::manager::httpd_dispatcher->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_dispatcher' -e 'pf::services::manager::httpd_dispatcher->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/httpd.dispatcher-docker-wrapper
 ExecStop=/bin/bash -c "docker stop httpd.dispatcher ; echo Stopped"
 Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/httpd.dispatcher-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_dispatcher_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_dispatcher_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_dispatcher_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_dispatcher_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-httpd.portal.service b/conf/systemd/packetfence-httpd.portal.service
index 8b7bb297f5..2cc511504b 100644
--- a/conf/systemd/packetfence-httpd.portal.service
+++ b/conf/systemd/packetfence-httpd.portal.service
@@ -9,7 +9,7 @@ Type=notify
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_portal' -e 'pf::services::manager::httpd_portal->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_portal' -e 'pf::services::manager::httpd_portal->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/httpd.portal-docker-wrapper
 ExecStop=/bin/bash -c "docker stop httpd.portal ; echo Stopped"
 TimeoutStopSec=60
@@ -17,8 +17,8 @@ Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/httpd.portal-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_portal_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_portal_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_portal_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_portal_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-httpd.webservices.service b/conf/systemd/packetfence-httpd.webservices.service
index cb8f01c0c5..dca05ec908 100644
--- a/conf/systemd/packetfence-httpd.webservices.service
+++ b/conf/systemd/packetfence-httpd.webservices.service
@@ -11,7 +11,7 @@ Type=notify
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_webservices' -e 'pf::services::manager::httpd_webservices->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::httpd_webservices' -e 'pf::services::manager::httpd_webservices->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/httpd.webservices-docker-wrapper
 ExecStop=/bin/bash -c "docker stop httpd.webservices ; echo Stopped"
 TimeoutStartSec=300
@@ -20,8 +20,8 @@ Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/httpd.webservices-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_webservices_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_webservices_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_webservices_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_httpd_webservices_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-ip6tables.service b/conf/systemd/packetfence-ip6tables.service
index ef9c7c9e03..7fad187f97 100644
--- a/conf/systemd/packetfence-ip6tables.service
+++ b/conf/systemd/packetfence-ip6tables.service
@@ -11,13 +11,13 @@ After=packetfence-iptables.service
 Type=simple
 ExecStart=/usr/local/pf/sbin/ip6tables_monitor.pl
 ExecStop=-/usr/bin/kill -TERM $MAINPID
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::ip6tables' -e 'pf::services::manager::ip6tables->new()->_stop()'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::ip6tables' -e 'pf::services::manager::ip6tables->new()->_stop()'
 KillSignal=TERM
 Restart=always
 RestartSec=5
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::ip6tables -Mpf::config -e 'pf::ip6tables::ip6tables_services_rules("add")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::ip6tables -Mpf::config -e 'pf::ip6tables::ip6tables_services_rules("add")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-iptables.service b/conf/systemd/packetfence-iptables.service
index 3096a086ab..49e84602c0 100644
--- a/conf/systemd/packetfence-iptables.service
+++ b/conf/systemd/packetfence-iptables.service
@@ -8,13 +8,13 @@ After=packetfence-base.target packetfence-config.service
 Type=simple
 ExecStart=/usr/local/pf/sbin/iptables_monitor.pl
 ExecStop=-/usr/bin/kill -TERM $MAINPID
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::iptables' -e 'pf::services::manager::iptables->new()->_stop()'
 KillSignal=TERM
 Restart=always
 RestartSec=5
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_services_rules("add")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_services_rules("add")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-kafka.service b/conf/systemd/packetfence-kafka.service
index 3de894cfe4..f9c5fc2b6b 100644
--- a/conf/systemd/packetfence-kafka.service
+++ b/conf/systemd/packetfence-kafka.service
@@ -8,15 +8,15 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=simple
 TimeoutStopSec=60
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::kafka' -e 'pf::services::manager::kafka->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::kafka' -e 'pf::services::manager::kafka->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/kafka-docker-wrapper
 ExecStop=/bin/bash -c "docker stop kafka ; echo Stopped"
 Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/kafka-systemd-notify.pid
 # iptables rules
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_kafka_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_kafka_rules("REMOVE")'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_kafka_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_kafka_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-keepalived.service b/conf/systemd/packetfence-keepalived.service
index 30dd3a1b45..87fb2c2520 100644
--- a/conf/systemd/packetfence-keepalived.service
+++ b/conf/systemd/packetfence-keepalived.service
@@ -10,14 +10,14 @@ StartLimitInterval=10
 Type=forking
 KillMode=process
 ExecStartPre=/bin/rm -f /usr/local/pf/var/run/keepalived.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::keepalived' -e 'pf::services::manager::keepalived->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::keepalived' -e 'pf::services::manager::keepalived->new()->generateConfig()'
 ExecStart=/usr/sbin/keepalived -f /usr/local/pf/var/conf/keepalived.conf --pid=/usr/local/pf/var/run/keepalived.pid
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_keepalived_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_keepalived_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_keepalived_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_keepalived_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-mariadb.service b/conf/systemd/packetfence-mariadb.service
index 011e2b095d..529d126f6a 100644
--- a/conf/systemd/packetfence-mariadb.service
+++ b/conf/systemd/packetfence-mariadb.service
@@ -37,8 +37,8 @@ LimitNOFILE=4096
 KillMode=control-group
 
 # iptables rules
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mariadb_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mariadb_rules("REMOVE")'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mariadb_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mariadb_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence-base.target
diff --git a/conf/systemd/packetfence-mysql-probe.service b/conf/systemd/packetfence-mysql-probe.service
index fb56a99660..e6d5fdda05 100644
--- a/conf/systemd/packetfence-mysql-probe.service
+++ b/conf/systemd/packetfence-mysql-probe.service
@@ -14,8 +14,8 @@ ExecStart=/usr/local/pf/sbin/mysql-probe
 Restart=on-failure
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mysql_prob_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mysql_prob_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mysql_prob_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_mysql_prob_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-netdata.service b/conf/systemd/packetfence-netdata.service
index 03c248c69f..2930343748 100644
--- a/conf/systemd/packetfence-netdata.service
+++ b/conf/systemd/packetfence-netdata.service
@@ -18,8 +18,8 @@ Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/netdata-systemd-notify.pid
 # iptables rules
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_netdata_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_netdata_rules("REMOVE")'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_netdata_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_netdata_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-ntlm-auth-api.service b/conf/systemd/packetfence-ntlm-auth-api.service
index d7ba65fd2f..f553ae2176 100644
--- a/conf/systemd/packetfence-ntlm-auth-api.service
+++ b/conf/systemd/packetfence-ntlm-auth-api.service
@@ -10,7 +10,7 @@ Type=forking
 TimeoutStopSec=60
 NotifyAccess=all
 LimitNOFILE=8192
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::ntlm_auth_api' -e 'pf::services::manager::ntlm_auth_api->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::ntlm_auth_api' -e 'pf::services::manager::ntlm_auth_api->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/ntlm-auth-api-docker-wrapper start
 ExecStop=/usr/local/pf/sbin/ntlm-auth-api-docker-wrapper stop
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfacct.service b/conf/systemd/packetfence-pfacct.service
index 35fc71303a..69f7fa5774 100644
--- a/conf/systemd/packetfence-pfacct.service
+++ b/conf/systemd/packetfence-pfacct.service
@@ -10,15 +10,15 @@ Type=notify
 TimeoutStopSec=60
 NotifyAccess=all
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfacct' -e 'pf::services::manager::pfacct->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfacct' -e 'pf::services::manager::pfacct->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfacct-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfacct ; echo Stopped"
 Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/pfacct-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfacct_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfacct_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfacct_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfacct_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-pfconnector-client.service b/conf/systemd/packetfence-pfconnector-client.service
index d6b0e1c975..7068194aae 100644
--- a/conf/systemd/packetfence-pfconnector-client.service
+++ b/conf/systemd/packetfence-pfconnector-client.service
@@ -13,7 +13,7 @@ Environment=LOG_LEVEL=INFO
 # same host. Set to 1/true for localhost:6060, or an explicit host:port. After
 # changing, run `systemctl daemon-reload` and restart the service.
 #Environment=CHISEL_PPROF=localhost:6061
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfconnector_client' -e 'pf::services::manager::pfconnector_client->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfconnector_client' -e 'pf::services::manager::pfconnector_client->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfconnector-client-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfconnector-client ; echo Stopped"
 Restart=always
diff --git a/conf/systemd/packetfence-pfconnector-server.service b/conf/systemd/packetfence-pfconnector-server.service
index 668a0ebd4c..06dc0a0925 100644
--- a/conf/systemd/packetfence-pfconnector-server.service
+++ b/conf/systemd/packetfence-pfconnector-server.service
@@ -12,15 +12,15 @@ Environment=LOG_LEVEL=INFO
 # Set to 1/true for localhost:6060, or an explicit host:port. After changing,
 # run `systemctl daemon-reload` and restart the service.
 #Environment=CHISEL_PPROF=localhost:6060
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfconnector_server' -e 'pf::services::manager::pfconnector_server->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfconnector_server' -e 'pf::services::manager::pfconnector_server->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfconnector-server-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfconnector-server ; echo Stopped"
 Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/pfconnector-server-systemd-notify.pid
 # iptables rules
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfconnector_server_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfconnector_server_rules("REMOVE")'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfconnector_server_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfconnector_server_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-pfcron.service b/conf/systemd/packetfence-pfcron.service
index 75c6419fac..786648024a 100644
--- a/conf/systemd/packetfence-pfcron.service
+++ b/conf/systemd/packetfence-pfcron.service
@@ -10,7 +10,7 @@ StartLimitIntervalSec=0
 Type=notify
 NotifyAccess=all
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfcron' -e 'pf::services::manager::pfcron->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfcron' -e 'pf::services::manager::pfcron->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfcron-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfcron ; echo Stopped"
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfdhcp.service b/conf/systemd/packetfence-pfdhcp.service
index deba9b6843..81bc9e40b7 100644
--- a/conf/systemd/packetfence-pfdhcp.service
+++ b/conf/systemd/packetfence-pfdhcp.service
@@ -16,8 +16,8 @@ ExecStart=/usr/local/pf/sbin/pfdhcp
 Restart=on-failure
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdhcp_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdhcp_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdhcp_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdhcp_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-pfdns-connector.service b/conf/systemd/packetfence-pfdns-connector.service
index 2375f241b5..6ed4756587 100644
--- a/conf/systemd/packetfence-pfdns-connector.service
+++ b/conf/systemd/packetfence-pfdns-connector.service
@@ -10,7 +10,7 @@ Type=notify
 TimeoutStopSec=60
 NotifyAccess=all
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfdns_connector' -e 'pf::services::manager::pfdns_connector->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfdns_connector' -e 'pf::services::manager::pfdns_connector->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfdns-connector-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfdns-connector ; echo Stopped"
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfdns.service b/conf/systemd/packetfence-pfdns.service
index 16524507dd..8773106fa3 100644
--- a/conf/systemd/packetfence-pfdns.service
+++ b/conf/systemd/packetfence-pfdns.service
@@ -10,13 +10,13 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=notify
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfdns' -e 'pf::services::manager::pfdns->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfdns' -e 'pf::services::manager::pfdns->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfdns -conf=/usr/local/pf/var/conf/pfdns.conf
 Restart=on-failure
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdns_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdns_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdns_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfdns_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-pfipset.service b/conf/systemd/packetfence-pfipset.service
index 67328ef811..59fea16fae 100644
--- a/conf/systemd/packetfence-pfipset.service
+++ b/conf/systemd/packetfence-pfipset.service
@@ -10,13 +10,13 @@ StartLimitBurst=3
 StartLimitInterval=60
 Type=notify
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfipset' -e 'pf::services::manager::pfipset->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfipset' -e 'pf::services::manager::pfipset->new()->generateConfig()'
 ExecStart=/bin/bash -c "export $(cat /usr/local/pf/var/conf/pfipset.env | xargs) && /usr/local/pf/sbin/pfhttpd run --adapter caddyfile -c /usr/local/pf/conf/caddy-services/pfipset.conf"
 Restart=on-failure
 Slice=packetfence.slice
 # Iptable
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfipset_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfipset_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfipset_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_pfipset_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-pfldapexplorer.service b/conf/systemd/packetfence-pfldapexplorer.service
index a6d25db1ad..857eb1e46a 100644
--- a/conf/systemd/packetfence-pfldapexplorer.service
+++ b/conf/systemd/packetfence-pfldapexplorer.service
@@ -11,7 +11,7 @@ TimeoutStopSec=60
 NotifyAccess=all
 LimitNOFILE=8192
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfldapexplorer' -e 'pf::services::manager::pfldapexplorer->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfldapexplorer' -e 'pf::services::manager::pfldapexplorer->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfldapexplorer-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfldapexplorer; echo Stopped"
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfpki.service b/conf/systemd/packetfence-pfpki.service
index 079a1d9e70..d42ff78357 100644
--- a/conf/systemd/packetfence-pfpki.service
+++ b/conf/systemd/packetfence-pfpki.service
@@ -9,7 +9,7 @@ Type=notify
 TimeoutStopSec=60
 NotifyAccess=all
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfpki' -e 'pf::services::manager::pfpki->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfpki' -e 'pf::services::manager::pfpki->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfpki-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfpki ; echo Stopped"
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfqueue-backend.service b/conf/systemd/packetfence-pfqueue-backend.service
index 40cf9cceda..99313b4270 100644
--- a/conf/systemd/packetfence-pfqueue-backend.service
+++ b/conf/systemd/packetfence-pfqueue-backend.service
@@ -11,7 +11,7 @@ TimeoutStopSec=60
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue_backend' -e 'pf::services::manager::pfqueue_backend->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue_backend' -e 'pf::services::manager::pfqueue_backend->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfqueue-backend
 TimeoutStopSec=30
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfqueue-go.service b/conf/systemd/packetfence-pfqueue-go.service
index ea836d4227..1060b1f9e4 100644
--- a/conf/systemd/packetfence-pfqueue-go.service
+++ b/conf/systemd/packetfence-pfqueue-go.service
@@ -12,7 +12,7 @@ TimeoutStopSec=60
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue_go' -e 'pf::services::manager::pfqueue_go->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue_go' -e 'pf::services::manager::pfqueue_go->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfqueue-go
 TimeoutStopSec=30
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfqueue-perl.service b/conf/systemd/packetfence-pfqueue-perl.service
index 35fcb6dcb1..b13a8f1f05 100644
--- a/conf/systemd/packetfence-pfqueue-perl.service
+++ b/conf/systemd/packetfence-pfqueue-perl.service
@@ -10,7 +10,7 @@ TimeoutStopSec=60
 NotifyAccess=all
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue_perl' -e 'pf::services::manager::pfqueue_perl->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfqueue_perl' -e 'pf::services::manager::pfqueue_perl->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfqueue
 TimeoutStopSec=30
 Restart=on-failure
diff --git a/conf/systemd/packetfence-pfsetacls.service b/conf/systemd/packetfence-pfsetacls.service
index 85b79c7271..d62c162571 100644
--- a/conf/systemd/packetfence-pfsetacls.service
+++ b/conf/systemd/packetfence-pfsetacls.service
@@ -9,7 +9,7 @@ StartLimitBurst=3
 StartLimitInterval=10
 Type=simple
 TimeoutStopSec=60
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfsetacls' -e 'pf::services::manager::pfsetacls->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfsetacls' -e 'pf::services::manager::pfsetacls->new()->generateConfig()'
 ExecStartPre=mkdir -p /usr/local/pf/var/conf/pfsetacls
 ExecStartPre=chown -R :1002 /usr/local/pf/var/conf/pfsetacls
 ExecStartPre=chmod 775 /usr/local/pf/var/conf/pfsetacls
diff --git a/conf/systemd/packetfence-pfsso.service b/conf/systemd/packetfence-pfsso.service
index be82261324..60fea647d0 100644
--- a/conf/systemd/packetfence-pfsso.service
+++ b/conf/systemd/packetfence-pfsso.service
@@ -11,7 +11,7 @@ TimeoutStopSec=60
 NotifyAccess=all
 LimitNOFILE=8192
 Environment=LOG_LEVEL=INFO
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfsso' -e 'pf::services::manager::pfsso->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::pfsso' -e 'pf::services::manager::pfsso->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/pfsso-docker-wrapper
 ExecStop=/bin/bash -c "docker stop pfsso ; echo Stopped"
 Restart=on-failure
diff --git a/conf/systemd/packetfence-proxysql.service b/conf/systemd/packetfence-proxysql.service
index b8c41df819..b9269fa612 100644
--- a/conf/systemd/packetfence-proxysql.service
+++ b/conf/systemd/packetfence-proxysql.service
@@ -9,14 +9,14 @@ Type=notify
 NotifyAccess=all
 ExecStartPre=/usr/local/pf/containers/docker-minimal-rules.sh
 ExecStartPre=/bin/rm -f /usr/local/pf/var/run/proxysql/proxysql.pid
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::proxysql' -e 'pf::services::manager::proxysql->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::proxysql' -e 'pf::services::manager::proxysql->new()->generateConfig()'
 ExecStart=/usr/local/pf/sbin/proxysql-docker-wrapper
 ExecStop=/bin/bash -c "docker stop proxysql ; docker container rm proxysql -f ; echo Stopped"
 TimeoutStopSec=60
 Restart=on-failure
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_proxysql_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_proxysql_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_proxysql_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_proxysql_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence-cluster.target
diff --git a/conf/systemd/packetfence-radiusd-acct.service b/conf/systemd/packetfence-radiusd-acct.service
index 0dc15fce6b..64e75fde8c 100644
--- a/conf/systemd/packetfence-radiusd-acct.service
+++ b/conf/systemd/packetfence-radiusd-acct.service
@@ -20,8 +20,8 @@ Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/radiusd-acct-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_acct_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_acct_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_acct_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_acct_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-radiusd-auth.service b/conf/systemd/packetfence-radiusd-auth.service
index d49e1ae9d9..caebd34dbe 100644
--- a/conf/systemd/packetfence-radiusd-auth.service
+++ b/conf/systemd/packetfence-radiusd-auth.service
@@ -12,14 +12,14 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
-ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n auth -Cxm -lstdout 
-ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n auth -fm 
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/sbin/freeradius -d /usr/local/pf/raddb  -n auth -Cxm -lstdout 
+ExecStart=/usr/sbin/freeradius -d /usr/local/pf/raddb  -n auth -fm 
 Restart=on-failure
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_auth_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_auth_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_auth_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_auth_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-radiusd-cli.service b/conf/systemd/packetfence-radiusd-cli.service
index 3821fdc204..689ea38b5d 100644
--- a/conf/systemd/packetfence-radiusd-cli.service
+++ b/conf/systemd/packetfence-radiusd-cli.service
@@ -20,8 +20,8 @@ Restart=on-failure
 Slice=packetfence.slice
 PIDFile=/usr/local/pf/var/run/radiusd-cli-systemd-notify.pid
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_cli_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_cli_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_cli_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_cli_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-radiusd-eduroam.service b/conf/systemd/packetfence-radiusd-eduroam.service
index 2a6c6be96e..9af37a29cb 100644
--- a/conf/systemd/packetfence-radiusd-eduroam.service
+++ b/conf/systemd/packetfence-radiusd-eduroam.service
@@ -11,14 +11,14 @@ Type=notify
 WatchdogSec=10
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
-ExecStartPre=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n eduroam -Cxm -lstdout
-ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n eduroam -fm 
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::radiusd' -e 'pf::services::manager::radiusd->new()->generateConfig()'
+ExecStartPre=/usr/sbin/freeradius -d /usr/local/pf/raddb  -n eduroam -Cxm -lstdout
+ExecStart=/usr/sbin/freeradius -d /usr/local/pf/raddb  -n eduroam -fm 
 Restart=on-failure
 Slice=packetfence.slice
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_eduroam_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_eduroam_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_eduroam_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_radiusd_eduroam_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/conf/systemd/packetfence-redis_ntlm_cache.service b/conf/systemd/packetfence-redis_ntlm_cache.service
index 04bcced7b5..8e735efcb0 100644
--- a/conf/systemd/packetfence-redis_ntlm_cache.service
+++ b/conf/systemd/packetfence-redis_ntlm_cache.service
@@ -7,7 +7,7 @@ After=packetfence-base.target packetfence-config.service packetfence-iptables.se
 [Service]
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_ntlm_cache' -e 'pf::services::manager::redis_ntlm_cache->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_ntlm_cache' -e 'pf::services::manager::redis_ntlm_cache->new()->generateConfig()'
 ExecStart=/usr/bin/redis-server /usr/local/pf/var/conf/redis_ntlm_cache.conf --daemonize no
 ExecStop=/bin/kill $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-redis_queue.service b/conf/systemd/packetfence-redis_queue.service
index b648a3a9d7..70d0af1ac3 100644
--- a/conf/systemd/packetfence-redis_queue.service
+++ b/conf/systemd/packetfence-redis_queue.service
@@ -7,7 +7,7 @@ After=packetfence-base.target packetfence-config.service packetfence-iptables.se
 [Service]
 StartLimitBurst=3
 StartLimitInterval=10
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_queue' -e 'pf::services::manager::redis_queue->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::redis_queue' -e 'pf::services::manager::redis_queue->new()->generateConfig()'
 ExecStart=/usr/bin/redis-server /usr/local/pf/var/conf/redis_queue.conf --daemonize no
 ExecStop=/bin/kill $MAINPID
 Restart=on-failure
diff --git a/conf/systemd/packetfence-snmptrapd.service b/conf/systemd/packetfence-snmptrapd.service
index d105f44bb4..e8d6afb896 100644
--- a/conf/systemd/packetfence-snmptrapd.service
+++ b/conf/systemd/packetfence-snmptrapd.service
@@ -8,14 +8,14 @@ Wants=packetfence-base.target packetfence-config.service packetfence-iptables.se
 StartLimitBurst=3
 StartLimitInterval=10
 Type=simple
-ExecStartPre=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::snmptrapd' -e 'pf::services::manager::snmptrapd->new()->generateConfig()'
+ExecStartPre=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5 '-Mpf::services::manager::snmptrapd' -e 'pf::services::manager::snmptrapd->new()->generateConfig()'
 ExecStart=/usr/sbin/snmptrapd -f -n -c /usr/local/pf/var/conf/snmptrapd.conf -C -A -Lf /usr/local/pf/logs/snmptrapd.log -p /usr/local/pf/var/run/snmptrapd.pid -On 
 ExecReload=/bin/kill -HUP $MAINPID
 Slice=packetfence.slice
 Restart=on-failure
 # iptables rules
-ExecStartPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_snmptrapd_rules("add")'
-ExecStopPost=/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_snmptrapd_rules("REMOVE")'
+ExecStartPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_snmptrapd_rules("add")'
+ExecStopPost=/usr/bin/perl -I/usr/local/pf/lib -I/usr/local/pf/lib_perl/lib/perl5/ -Mpf::iptables -Mpf::config -e 'pf::iptables::iptables_snmptrapd_rules("REMOVE")'
 
 [Install]
 WantedBy=packetfence.target
diff --git a/t/venom/scenarios/cli_login/playbooks/configure.yml b/t/venom/scenarios/cli_login/playbooks/configure.yml
index 67fbfae99a..b65f191999 100644
--- a/t/venom/scenarios/cli_login/playbooks/configure.yml
+++ b/t/venom/scenarios/cli_login/playbooks/configure.yml
@@ -221,7 +221,7 @@
         WatchdogSec=20
         StartLimitBurst=3
         StartLimitInterval=10
-        ExecStart=/usr/sbin/radiusd -d /usr/local/pf/raddb  -n test -fm
+        ExecStart=/usr/sbin/freeradius -d /usr/local/pf/raddb  -n test -fm
         Restart=on-failure
     when: ansible_facts['os_family'] == "RedHat"
 
